Add web-based OAuth authentication

brendandburns opened the pull request:
Also add /approve to the actions

Builds off of #1908 so merge that first.

krzyzacy commented on mungegithub/mungers/bulk-lgtm.go:
```diff
@@ -17,12 +17,19 @@ limitations under the License.
+	"strings"
 	"sync"
 
+	"golang.org/x/oauth2"
```
you want to vendor the lib :-)

brendandburns replied:
It's already there afaik:

https://github.com/kubernetes/test-infra/tree/master/vendor/golang.org/x/oauth2

krzyzacy replied:
you are 100% right..

krzyzacy replied:
oh, yup... you need to add oauth2 to the [BUILD file](https://github.com/kubernetes/test-infra/blob/master/mungegithub/mungers/BUILD#L90)

krzyzacy commented:
@k8s-bot bazel test this

apelisse commented:
Looks good to me.

Should we remove the part about approval though, because of https://github.com/kubernetes/test-infra/issues/1744

brendandburns commented:
@apelisse anything more needed here? looks like Travis is stuck...

apelisse commented:
Yeah travis is fine, it completed successfully, just didn't update the status.
Could you remove the part about approval, as LGTM now auto-approves?

apelisse commented:
@brendanburns

apelisse commented:
@brendandburns

brendandburns commented:
@apelisse removed `/approve`, added `/release-note-none`

Also moved to storing the oauth token in a cookie.

Please take another look.

Thanks!
--brendan

apelisse commented:
Sorry for not seeing that earlier!

/lgtm

brendandburns commented:
@apelisse what is needed to merge this?

Thanks!

apelisse commented:
Nothing. test-infra is on a manual merge basis...
